src/pages/cohort/[cohortSlug]/[slug]/[version]/index.jsx

```diff
@@ -179,7 +179,28 @@ function Dashboard() {
         });
       });
   };
+
+  const checkNavigationAvailability = () => {
+    if (allSubscriptions) {
+      const currentSessionSubs = allSubscriptions?.filter((sub) => sub.academy?.id === cohortSession?.academy?.id);
+      const cohortSubscriptions = currentSessionSubs?.filter((sub) => sub.selected_cohort_set?.cohorts.some((cohort) => cohort.id === cohortSession.id));
+      if (cohortSubscriptions.length === 0) router.push('/choose-program');
```

Comment:
Show them an error saying they don't have access to this program and to contact support if they believe there is a mistake